From 233e1768fff838b70b9e942c4a5eee60e57c54d4 Mon Sep 17 00:00:00 2001
From: orderedstereographic
 <161899146+orderedstereographic@users.noreply.github.com>
Date: Wed, 22 Oct 2025 19:29:53 -0500
Subject: [PATCH] xpadneo: Remove deprecated ida_simple_xx() usage

Update to use ida_alloc() and ida_free() in place of the deprecated
ida_simple_get() and ida_simple_remove() calls. Linux has removed this
API in the v6.18 release candidates. ida_simple_xx() has been deprecated since
v5.10 and the new API was introduced in v4.19.

Closes: https://github.com/atar-axis/xpadneo/pull/562
---
 hid-xpadneo/src/hid-xpadneo.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/hid-xpadneo/src/hid-xpadneo.c b/hid-xpadneo/src/hid-xpadneo.c
index ac3e135f..77c741bb 100644
--- a/hid-xpadneo/src/hid-xpadneo.c
+++ b/hid-xpadneo/src/hid-xpadneo.c
@@ -1223,7 +1223,7 @@ static int xpadneo_probe(struct hid_device *hdev, const struct hid_device_id *id
 	if (xdata == NULL)
 		return -ENOMEM;
 
-	xdata->id = ida_simple_get(&xpadneo_device_id_allocator, 0, 0, GFP_KERNEL);
+	xdata->id = ida_alloc(&xpadneo_device_id_allocator, GFP_KERNEL);
 	xdata->quirks = id->driver_data;
 
 	xdata->hdev = hdev;
@@ -1328,7 +1328,7 @@ static int xpadneo_probe(struct hid_device *hdev, const struct hid_device_id *id
 static void xpadneo_release_device_id(struct xpadneo_devdata *xdata)
 {
 	if (xdata->id >= 0) {
-		ida_simple_remove(&xpadneo_device_id_allocator, xdata->id);
+		ida_free(&xpadneo_device_id_allocator, xdata->id);
 		xdata->id = -1;
 	}
 }
